semantics/src/main/scala/aqua/semantics/rules/types/TypesInterpreter.scala

```diff
@@ -58,6 +58,12 @@ class TypesInterpreter[S[_], X](using
         report.error(token, s"Unresolved type").as(None)
     }
 
+  def resolveNamedType(token: TypeToken[S]): State[X, Option[Type]] =
+    resolveType(token).flatMap(_.flatTraverse {
```

Comment:
nit: I believe we should move towards `OptionT` because we have `F[Option[A]]` almost all the time and it produces a lot of such `flatMap(_.flatTraverse(` chains. But I am not insisting on rewriting this right now